src/test/java/com/microsoft/sqlserver/jdbc/AETest.java

```diff
@@ -0,0 +1,499 @@
+     * 
+     * @throws Exception
+     *         when an error occurs
+     */
+    @BeforeAll
+    public static void setUp() throws Exception {
+        connectionStringAE = TestUtils.addOrOverrideProperty(connectionString, "columnEncryptionSetting",
+                ColumnEncryptionSetting.Enabled.toString());
+
+        String enclaveAttestationUrl = TestUtils.getConfiguredProperty("enclaveAttestationUrl");
+        connectionStringAE = TestUtils.addOrOverrideProperty(connectionStringAE, "enclaveAttestationUrl",
+                (null != enclaveAttestationUrl) ? enclaveAttestationUrl : "http://blah");
+
+        String enclaveAttestationProtocol = TestUtils.getConfiguredProperty("enclaveAttestationProtocol");
+        connectionStringAE = TestUtils.addOrOverrideProperty(connectionStringAE, "enclaveAttestationProtocol",
+                (null != enclaveAttestationProtocol) ? enclaveAttestationProtocol : AttestationProtocol.HGS.toString());
+
+        try (SQLServerConnection con = PrepUtil.getConnection(connectionStringAE)) {
+            isAEv2 = TestUtils.isAEv2(con);
+        } catch (SQLException e) {
+            isAEv2 = false;
+        } catch (Exception e) {
+            fail(TestResource.getResource("R_unexpectedErrorMessage") + e.getMessage());
+        }
+
+        // reset logging to avoid severe logs due to negative testing
+        LogManager.getLogManager().reset();
+
+        dsLocal = new SQLServerDataSource();
+        AbstractTest.updateDataSource(connectionStringAE, dsLocal);
+
+        dsXA = new SQLServerXADataSource();
+        AbstractTest.updateDataSource(connectionStringAE, dsXA);
+
+        dsPool = new SQLServerConnectionPoolDataSource();
+        AbstractTest.updateDataSource(connectionStringAE, dsPool);
+    }
+
+    /**
+     * Tests basic connection.
+     * 
+     * @throws SQLException
+     *         when an error occurs
+     */
+    @Test
+    public void testBasicConnection() throws SQLException {
+        org.junit.Assume.assumeTrue(isAEv2);
+
+        try (Connection con1 = dsLocal.getConnection(); Connection con2 = dsXA.getConnection();
+                Connection con3 = dsPool.getConnection();
+                Connection con4 = PrepUtil.getConnection(connectionStringAE)) {
+            if (TestUtils.isAEv2(con1)) {
+                verifyEnclaveEnabled(con1);
+            }
+            if (TestUtils.isAEv2(con2)) {
+                verifyEnclaveEnabled(con2);
+            }
+            if (TestUtils.isAEv2(con3)) {
+                verifyEnclaveEnabled(con3);
+            }
+            if (TestUtils.isAEv2(con4)) {
+                verifyEnclaveEnabled(con4);
+            }
+        }
+    }
+
+    /**
+     * Tests invalid connection property combinations.
+     */
+    @Test
+    public void testInvalidProperties() {
+        org.junit.Assume.assumeTrue(isAEv2);
+
+        // enclaveAttestationUrl and enclaveAttestationProtocol without "columnEncryptionSetting"

```

Comment:
Please create a copy of the original connection string to not modify it.

Reply:
this doesn't change it, it returns a copy